### ISSUE TITLE
Filter 'get_the_excerpt' is called with a missing argument

### ISSUE THREAD

**Issue by mboynes** [Automattic/jetpack#9209]
I experienced a fatal error on a site from [class.media-summary.php's usage of the `get_the_excerpt` filter](https://github.com/Automattic/jetpack/blob/d6846b492329cc6436ee9db93d8a44b32476e2f7/_inc/lib/class.media-summary.php#L314). The `'get_the_excerpt'` filter is being applied with only one argument, not matching [how core uses that filter](https://github.com/WordPress/WordPress/blob/921e131eae45801b8fdb1ecfceb5d7839fdfd509/wp-includes/post-template.php#L402-L411).

Since Jetpack isn't using a core filter as it is intended, it's relatively easy for themes and plugins to encounter a conflict with Jetpack, causing a site to encounter this fatal error.

Relates to #8510, which would render this moot, but is currently on hold.

**Comment by jeherve** [comment]
Is your site on PHP 7.2? If so, I would be inclined to close this as duplicate of #8420.

Thank you!

**Comment by mboynes** [comment]
I am running PHP 7.2, though it's also a fatal error in PHP 7.1. Prior PHP versions handle 'missing argument' errors as `PHP Warning`s. Regardless, a core filter is being applied without matching how core uses it, and that is bound to cause compatibility issues.

This is not a duplicate of #8420; they just happen to both affect the same block of code.

Thanks!

**Comment by jeherve** [comment]
Could you let me know what other plugins you run on your site, so we can easily reproduce the fatal on our end? That would help when testing the fix offered in #8510, and make sure it fixes your issue as well.

Thank you!

**Comment by mboynes** [comment]
The code where I encountered this is custom. To replicate, here's a simple example...

```php
function my_excerpt_function( $text, $post ) {
    return $text;
}
add_filter( 'get_the_excerpt', 'my_excerpt_function', 10, 2 );
```

The code in #8510 renders the problem moot, since it removes the core filter application entirely.
